applications/dedot_typink.md

```diff
@@ -0,0 +1,187 @@
+
+- https://github.com/sinzii
+- https://github.com/1cedrus
+
+## Development Status :open_book:
+
+Both Dedot and Typink have been developed through the W3F Grants Program, where we've successfully completed multiple milestones that established the foundation for these tools:
+
+- **Dedot** serves as a foundation for ink! contract interactions in JavaScript/TypeScript, with its lightweight design and type-safe approach reducing development errors and improving code quality for ink! developers.
+
+- **Typink** provides fully type-safe React hooks for ink! contract interactions and a built-in CLI that helps developers onboard and set up projects easier and faster than before.
+
+This proposal builds upon these foundations to address the next set of challenges in the ink! ecosystem: supporting the latest protocol changes (ink! v6) with `pallet-revive` integration, and further improve the toolings to streamline the developer experience and onboarding process.
+
+## Development Roadmap :nut_and_bolt:
+
+### Overview
+
+- **Total Estimated Duration:** 5 months
+- **Full-Time Equivalent (FTE):** 1.5 FTE
+- **Total Costs:** 50,000 USD
+
+### Milestone 1: Dedot - pallet-revive Support & ink! v6 Integration
+
+- **Estimated duration:** 2.5 months
+- **FTE:** 1.5
+- **Costs:** 25,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and **online documentation** for the new features on https://dedot.dev. This will include comprehensive examples of how developers can interact with ink! v6 contracts via dedot. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Article & Communication | We will publish an article introducing the changes and features implemented in this milestone and post regular updates on Polkadot Forum and social media. |
+| 1. | pallet-revive support | - Add support for 20-byte (H160) address format used in Ethereum-compatible environments<br />- Implement support for signing transactions with EVM wallets like MetaMask and SubWallet<br />- Implement comprehensive support for sending contract queries, submitting contract transactions, and deploying contracts.<br />- Develop functionality to listen to and decode contract events from pallet-revive<br /> |
+| 2. | ink! v6 integration | - Add support for ink! v6 while maintaining compatibility with v4/v5<br />- Generate Typescript Types & APIs using both traditional Rust metadata & Solidity metadata format supported by ink! v6<br />- Enable auto-suggestions and IntelliSense for generated types and APIs<br />- Ensure seamless interaction between ink! v6 contracts and pallet-revive |
+
+### Milestone 2: Typink - ink! v6 Integration, Wallet Support & CLI Improvements
```

Comment:
Curious whether the order provided within the breakdown of this milestone indicates the priority or an idea of the plan of attack you are thinking about following to work and deliver the milestone.

If that is the case my suggestion would be doing point `5.` right after the integration with ink!v6 on point `1.` Maybe even draw a line with these two points, and then tackle `txToaster` and `create-typink`.

Reply:
Overall, this is the list of tasks we plan to work on, but the order should remain flexible during implementation so we can adjust priorities as needed if certain tasks become more important as we progress.

Regarding point 5, we planned to place it at the end to document all completed tasks for the milestone, we will also update the documentation progressively as each feature becomes available in the toolkit/library. This way, developers can start using the new features right away with clear guidance and documentation.

Let me know if that addresses your concerns, thanks!

Reply:
Alright, all good.